src/App.tsx

```diff
@@ -1,24 +1,197 @@
+import { Header } from './components/Header';
+import { Footer } from './components/Footer';
+import { Notification } from './components/Notification';
+import { SortType } from './types/SortType';
+import { prepareTodos } from './utils/prepareTodos';
+import { ErrorType } from './types/ErrorType';
 
-const USER_ID = 0;
+const USER_ID = 6358;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorType, setErrorType] = useState(ErrorType.NONE);
+  const [sortType, setSortType] = useState<SortType>(SortType.ALL);
+  const [isLoading, setIsLoading] = useState(false);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [processingTodos, setProcessingTodos] = useState<number[]>([]);
+
+  const isTodos = todos.length !== 0;
+  const activeTodosAmount = todos.filter(todo => !todo.completed).length;
+  const isCompletedTodos = todos.some(todo => todo.completed);
+  const completedTodos = todos.filter(todo => todo.completed);
+  const isAllCompleted = todos.length === completedTodos.length;
+
+  const autoCloseNotification = useCallback(() => {
+    setTimeout(() => {
+      setErrorType(ErrorType.NONE);
+    }, 3000);
+  }, []);
+
+  const addErrorMessage = (errorMessage: ErrorType) => {
+    setErrorType(errorMessage);
+    autoCloseNotification();
+  };
+
+  const fetchedTodos = async () => {
+    try {
+      setErrorType(ErrorType.NONE);
+      const data = await getTodos(USER_ID);
+
+      setTodos(data);
+      setIsLoading(false);
+    } catch (error) {
+      // eslint-disable-next-line no-console
+      console.log(error);
+      addErrorMessage(ErrorType.GET);
+    }
+  };
+
+  useEffect(() => {
+    fetchedTodos();
+  }, []);
+
+  const handleSortTodos = useCallback((sort: SortType) => {
+    setSortType(sort);
+  }, []);
+
+  const handleCloseNotification = useCallback(() => {
+    setErrorType(ErrorType.NONE);
+  }, []);
+
+  const visibleTodos = useMemo(() => (
+    prepareTodos(todos, sortType)
+  ), [todos, sortType]);
+
+  const handleAddTodo = useCallback(async (
+    title: string, userId: number,
+  ) => {
+    const newTempTodo = {
+      id: 0,
+      userId: USER_ID,
+      title,
+      completed: false,
+    };
+
+    try {
+      setIsLoading(true);
+      setTempTodo(newTempTodo);
+
+      await addTodo(title, userId);
+      await fetchedTodos();
+    } catch (error) {
+      addErrorMessage(ErrorType.ADD);
+    } finally {
+      setTempTodo(null);
+      setIsLoading(false);
+    }
+  }, []);
+
+  const handleDeleteTodo = useCallback(async (todoId: number) => {
+    try {
+      setProcessingTodos(prevId => ([...prevId, todoId]));
+      await deleteTodo(todoId);
+      await fetchedTodos();
+    } catch (error) {
+      addErrorMessage(ErrorType.DELETE);
+    } finally {
+      setProcessingTodos([]);
+    }
+  }, []);
+
+  const handleClearCompleted = () => {
+    completedTodos.forEach(todo => {
+      handleDeleteTodo(todo.id);
+    });
```

Comment:
As a suggestion, you can check for completed todos directly here and if completed pass it to deleteTodo function.

Reply:
`todos.forEach(todo => {
if (todo.completed) {
handleDeleteTodo(todo.id)}
})`

Reply:
> As a suggestion, you can check for completed todos directly here and if completed pass it to deleteTodo function.

I'd want to clarify, do you mean like this?

```  
 todos.forEach(todo => {
   if (todo.completed) {
     handleDeleteTodo(todo.id);
   }
 });
```